paasta_tools/paasta_remote_run.py

```diff
@@ -109,9 +117,107 @@ def extract_args(args):
     return (system_paasta_config, service, cluster, soa_dir, instance, instance_type)
 
 
+def paasta_to_task_config_kwargs(
+        service,
+        instance,
+        cluster,
+        system_paasta_config,
+        instance_type='paasta_native',
+        soa_dir=DEFAULT_SOA_DIR,
+        config_overrides=None):
+    native_job_config = load_paasta_native_job_config(
+        service,
+        instance,
+        cluster,
+        soa_dir=soa_dir,
+        instance_type=instance_type,
+        config_overrides=config_overrides
+    )
+
+    image = native_job_config.get_docker_url()
+    docker_parameters = [
+        {'key': param['key'], 'value': param['value']}
+        for param in native_job_config.format_docker_parameters()
+    ]
+    # network = native_job_config.get_mesos_network_mode()
+
+    docker_volumes = native_job_config.get_volumes(
+        system_volumes=system_paasta_config.get_volumes()
+    )
+    volumes = [
+        {
+            'container_path': volume['containerPath'],
+            'host_path': volume['hostPath'],
+            'mode': volume['mode'].upper(),
+        }
+        for volume in docker_volumes
+    ]
+    cmd = native_job_config.get_cmd()
+    uris = system_paasta_config.get_dockercfg_location()
+    cpus = native_job_config.get_cpus()
+    mem = native_job_config.get_mem()
+    disk = native_job_config.get_disk(10)
+
+    kwargs = {
+        'image': str(image),
+        'cmd': cmd,
+        'cpus': cpus,
+        'mem': float(mem),
+        'disk': float(disk),
+        'volumes': volumes,
+        # 'ports': None,
+        # 'cap_add'
+        # 'ulimit'
+        'uris': [uris],
+        'docker_parameters': docker_parameters
+    }
+
+    config_hash = get_config_hash(
+        kwargs,
+        force_bounce=native_job_config.get_force_bounce(),
+    )
+
+    kwargs['name'] = str(compose_job_id(
+        service,
+        instance,
+        git_hash=get_code_sha_from_dockerurl(image),
+        config_hash=config_hash,
+        spacer=MESOS_TASK_SPACER,
+    ))
+
+    return kwargs
+
+
+def build_executor_stack(
+        service,
+        instance,
+        run_id,  # TODO: move run_id into task identifier?
+        system_paasta_config,
+        framework_staging_timeout):
+    mesos_address = '{}:{}'.format(
+        mesos_tools.get_mesos_leader(), mesos_tools.MESOS_MASTER_PORT
+    )
+
+    taskproc_config = system_paasta_config.get('taskproc')
+    # TODO: implement DryRunExecutor?
+    mesos_executor = MesosExecutor(
+        role=taskproc_config.get('role', taskproc_config['principal']),
+        principal=taskproc_config['principal'],
+        secret=taskproc_config['secret'],
+        mesos_address=mesos_address,
+        framework_name="paasta-remote %s %s %s" % (
```

Comment:
rather than using a random string for the runid, why not just use millis in the date field?